### ISSUE TITLE
Failed to kekulize aromatic bonds in OBMol::PerceiveBondOrders

### ISSUE THREAD

**Issue by roshkjr** [openbabel/openbabel#2567]
## Environment Information
Open Babel version: 3.1.0
Operating system and version:macOS Ventura (13.2)

## Expected Behavior
Create OBMol object without warning

## Actual Behavior
Created OBMol object with "Failed to kekulize aromatic bonds in OBMol::PerceiveBondOrders" warning

## Steps to Reproduce
import gemmi
from openbabel import openbabel as ob
path = "path to mmcif file containing protein structure" 
st = gemmi.read_structure(path, merge_chain_parts=True)
del st[1:]
groups = gemmi.MmcifOutputGroups(True)
groups.group_pdb = True
cif_block = st.make_mmcif_block(groups)
atom_site = cif_block.get_mmcif_category("_atom_site.")
last_res_id = None
last_res_name = None
last_chain_id = None
chain_num = 0
res = None

mol = ob.OBMol()
mol.SetChainsPerceived()
mol.BeginModify()

for i in range(len(atom_site["id"])):
    current_res_id = _get_res_id(atom_site, i)
    ins_code = _get_ins_code(atom_site, i)

    if last_chain_id != atom_site["auth_asym_id"][i]:
        chain_num += 1
        last_chain_id = atom_site["auth_asym_id"][i]
    
    if (
        current_res_id != last_res_id
        or atom_site["auth_asym_id"][i] != last_chain_id
        or atom_site["label_comp_id"][i] != last_res_name
       ):

        last_res_id = current_res_id
        last_res_name = atom_site["label_comp_id"][i]
        res = mol.NewResidue()
        res.SetChainNum(chain_num)
        res.SetNum(str(last_res_id))
        res.SetName(atom_site["label_comp_id"][i])
        res.SetInsertionCode(ins_code)

    atom = mol.NewAtom(int(atom_sites["id"][i]))
    atom.SetVector(
        float(atom_sites["Cartn_x"][i]),
        float(atom_sites["Cartn_y"][i]),
        float(atom_sites["Cartn_z"][i]),
    )
        
    type_symbol = atom_sites["type_symbol"][i]
    element = f'{type_symbol[0]}{type_symbol[1].lower()}' if len(type_symbol) == 2 else type_symbol

    atomic_num = ob.GetAtomicNum(element)
    atom.SetAtomicNum(atomic_num)
    atom.SetFormalCharge(_format_formal_charge(atom_sites, i))
    res.AddAtom(atom)
    res.SetHetAtom(atom, atom_sites["group_PDB"][i] == "HETATM")
    res.SetSerialNum(atom, int(atom_sites["id"][i]))

    resdat = ob.OBResidueData()
    resdat.AssignBonds(mol)
    mol.ConnectTheDots()
    mol.PerceiveBondOrders()

The OBMol object is then used to calculate protein-ligand interactions using arpeggio package. While using Open Babel 2.4.1, the warning was not present. But when updated to 3.1.0 version the warning appeared for several proteins from wwwPDB and mismatch in several protein-ligand interactions calculated. 

I would like to now if there is any update in the way protein bond orders perceived? Is the warning can be ignored for my use-case (generating OBMol object from scratch using mmcif files)?

**Comment by JorgeG94** [comment]
> From the release notes of 3.0.0:
> 
> > Code for handling implicit hydrogens and kekulization has been entirely replaced. As well as being accurate, the new approach is much faster.
> 
> It is most likely safe to ignore the warning, but I would check the results to see that the structures look like you expect.

can the warning be silenced?

**Comment by fredrikw** [comment]
I don't think you could silence only that warning, but you can silence all warnings with `ob.obErrorLog.SetOutputLevel(0)` (see https://stackoverflow.com/questions/50419371/how-to-disable-logged-warnings-in-pybel and https://openbabel.org/api/3.0/classOpenBabel_1_1OBMessageHandler.shtml)

**Comment by husakm** [comment]
Maybe the molecule  is protonized ? OB fails to assign correct atom types e.g. for cycle with protonized N ...

**Comment by ghutchis** [comment]
This is not a bug IMHO. It's giving a chemically relevant warning.
